src/util/bip32.rs

```diff
@@ -187,6 +202,22 @@ impl serde::Serialize for ChildNumber {
     }
 }
 
+/// Parse a BIP32 derivation path.
+pub fn parse_derivation_path(path: &str) -> Result<Vec<ChildNumber>, Error> {
+    let mut parts = path.split("/");
+    // First parts must be `m`.
+    if parts.next().unwrap() != "m" {
+        return Err(Error::InvalidDerivationPathFormat);
+    }
+
+    // Empty parts are a format error.
+    if parts.clone().any(|p| p.len() == 0) {
```

Comment:
This check seems unnecessary -- `from_str` on `ChildNumber` will already fail if you give it an empty string as input.

Reply:
Yeah, but I personally didn't like that you would get the `InvalidChildNumber` error if you have an empty part instead of `InvalidDerivationPath`.

Reply:
I see both sides, but since it's simpler to not do the check here, let's not do the check.

Reply:
Done.